File: 409_longest_palindrome/longestPalindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int res = 0;
        int n = s.length();

        Set<Character> visited = new HashSet<>();

        for (int i = 0; i < n; i++) {
            char c = s.charAt(i);

            if (visited.contains(c)) {
                res += 2;
                visited.remove(c);
            } else {
                visited.add(c);
            }
        }

        if (!visited.isEmpty()) res++;

        return res;
    }
}

File: 409_longest_palindrome/longestPalindrome.cpp
class Solution {
    public:
        int longestPalindrome(string s) {
            int res = 0;
    
            set<char> visited;
    
            for (char c : s) {
                if (visited.find(c) != visited.end()) {
                    res += 2;
                    visited.erase(c);
                } else {
                    visited.insert(c);
                }
            }
    
            if (!visited.empty()) res++;
    
            return res;
        }
    };